routes/dao_coin_exchange.go

```diff
@@ -74,16 +82,115 @@ func (fes *APIServer) GetDAOCoinLimitOrders(ww http.ResponseWriter, req *http.Re
 		return
 	}
 
+	txnStatus := requestData.TxnStatus
+	if txnStatus == "" {
+		txnStatus = TxnStatusUnconfirmed
+	} else if txnStatus != TxnStatusUnconfirmed &&
+		txnStatus != TxnStatusConfirmed {
+
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("GetDAOCoinLimitOrders: Invalid TxnStatus: %v. Options "+
+				"are {unconfirmed, confirmed}.", txnStatus),
+		)
+		return
+	}
+
 	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
 	if err != nil {
 		_AddInternalServerError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Problem fetching utxoView: %v", err))
 		return
 	}
+	if txnStatus == TxnStatusConfirmed {
+		utxoView = lib.NewUtxoView(
+			fes.backendServer.GetBlockchain().DB(),
+			fes.Params,
+			nil,
+			fes.backendServer.GetBlockchain().Snapshot(),
+			nil)
+	}
+
+	// If they requested a specific set of OrderIds then fetch them directly
+	if len(requestData.OrderIds) > 0 {
```

Comment:
This kinda feels like we're overloading this endpoint now. Any reason we can't break it into a separate one?